encodings/dict/src/compute/compare.rs

```diff
@@ -24,25 +19,22 @@ impl CompareKernel for DictVTable {
         if lhs.values().len() > lhs.codes().len() {
             return Ok(None);
         }
+
         // If the RHS is constant, then we just need to compare against our encoded values.
         if let Some(rhs) = rhs.as_constant() {
             let compare_result = compare(
                 lhs.values(),
                 ConstantArray::new(rhs, lhs.values().len()).as_ref(),
                 operator,
             )?;
-            return if operator == Operator::Eq {
```

Comment:
I don't think you can delete this, it has nothing to do with nullability but instead with whether we enforce that dictionaries are unique (which we don't)